pyccel/codegen/printing/ccode.py

```diff
@@ -422,7 +430,29 @@ def _print_IsNot(self, expr):
 
             return '{} != {}'.format(lhs, rhs)
         else:
-            raise NotImplementedError
+            errors.report(PYCCEL_RESTRICTION_IS_ISNOT,
+                          symbol=expr, severity='fatal')
+
+    def _print_Is(self, expr):
```

Comment:
Is it possible to combine the two functions `_print_Is` and `_print_IsNot` into one, similarly to what is done in parser/semantic.py? This means you only need to update one function when making changes and guarantees that the two functions use the same method. You could pass the printed form of the operator (`==` for `is`, `!=` for `is not`) as an argument to avoid requiring several if/else blocks

Reply:
This would be useful indeed.